### PR TITLE
Update branch list for 1.12, remove 1.8

### PR DESCRIPTION
**Pull request by markan**
Signed-off-by: Mark Anderson <manderson@hashicorp.com>

### CODE REVIEW

File: .release/ci.hcl [hunk 1]
```diff
@@ -1,20 +1,20 @@
 schema = "1"
 
-project "consul" {
+project "consul-enterprise" {
```
Comment (kisunji) [inline .release/ci.hcl:3]: Is this intentional?

File: .release/ci.hcl [hunk 2]
```diff
@@ -1,20 +1,20 @@
...
   team = "consul core"
   slack {
     # feed-consul-ci
     notification_channel = "C9KPKPKRN"
   }
   github {
     organization = "hashicorp"
-    repository = "consul"
+    repository = "consul-enterprise"
```
Comment (kisunji) [inline .release/ci.hcl:11]: Same here

**Review by kisunji — APPROVED** [review]

File: .release/ci.hcl (hunk 1 above)
Comment (rboyer) [reply .release/ci.hcl]: This is likely wrong.

Comment (markan) [reply .release/ci.hcl]: Good catch, fixed.

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/628612.

**Pull request merged**